### PR TITLE
Improvements to Combination Lock and Two Factor.

### PR DESCRIPTION
**Pull request by samfundev**
- Aligned text to buttons.
- Fixed keeping the code within 0-19.
- Added Twitch Plays Support.
- Added logging.
- Cleaned up code.
- Added a log message when a Two Factor is added to the bomb.

### CODE REVIEW

File: Assets/Editor/Resources/ModConfig.asset [hunk 1]
```diff
@@ -14,4 +14,5 @@ MonoBehaviour:
   id: combination_lock
   title: Combination Lock
   version: 1
-  outputFolder: build
+  outputFolder: C:\Program Files (x86)\Steam\steamapps\common\Keep Talking and Nobody
```
Comment (timtmok) [inline Assets/Editor/Resources/ModConfig.asset]: I'm not keen on a hard coded path here especially when it's tied to Windows.

File: Assets/Scripts/CombinationLockModule.cs [hunk 2]
```diff
@@ -173,86 +191,98 @@ private void GeneratePassCode()
...
-
-	private static int GetNumberOfModules(KMBombInfo bombInfo)
-	{
-		return bombInfo == null ? 3 : bombInfo.GetModuleNames().Count;
+		int i;
+		return int.TryParse(input, out i) ? (int?) i : null;
 	}
 
-	private static int GetNumberOfSolvedModules(KMBombInfo bombInfo)
+	public IEnumerator ProcessTwitchCommand(string command)
```
Comment (timtmok) [inline Assets/Scripts/CombinationLockModule.cs:253]: Where do I find Twitch Plays? Do I need anything else to compile this or to test it?

**Review by timtmok — COMMENTED** [review]
Smaller commits would be easier to review bit by bit. I'll cherry-pick the change and try to build when I can. I don't think any changes will be necessary for now. I'll update this once I try building it.

File: Assets/Scripts/CombinationLockModule.cs (hunk 2 above)
Comment (timtmok) [reply Assets/Scripts/CombinationLockModule.cs:253]: I've managed to build it and can see the test harness supports it.

Comment (samfundev) [reply Assets/Scripts/CombinationLockModule.cs:253]: ProcessTwitchCommand is a method that a module can expose for adding Twitch Plays support. Normally you would need to launch the game with Twitch Plays to test it but the harness has support for testing it.

**Pull request merged**